src/Common/src/System/Net/Security/Unix/SafeFreeSslCredentials.cs

```diff
@@ -79,8 +85,22 @@ public SafeFreeSslCredentials(X509Certificate certificate, SslProtocols protocol
                     throw new NotSupportedException(SR.net_ssl_io_no_server_cert);
                 }
 
-                _certHandle = Interop.Crypto.X509Duplicate(cert.Handle);
+                _certHandle = Interop.Crypto.X509Duplicate(cert.Handle);                
                 Interop.Crypto.CheckValidOpenSslHandle(_certHandle);
+
+                using (var chain = new X509Chain())
+                {
+                    if (chain.Build(cert))
+                    {
+                        var ar = new SafeX509Handle[chain.ChainElements.Count - 1];
+                        for (int i = 0; i < ar.Length; i++)
+                        {
+                            ar[i] = Interop.Crypto.X509Duplicate(chain.ChainElements[1 + i].Certificate.Handle);
+                            Interop.Crypto.CheckValidOpenSslHandle(ar[i]);
+                        }
+                        _certChainHandles = ar;
```

Comment:
For this part of the loop, it'd be nice if it a) looked more like https://github.com/stephentoub/corefx/blob/0b8dfc5774f66b8538e07813113d6c962a8d5775/src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertificateProvider.cs#L102-L116 and/or b) was factored out to be the same code, if possible.

Since they're doing different approaches, b might not be possible.  But this should be of length chain.Count - 2 (neither the leaf nor the root are supposed to be sent as "extra").